cmd/osbuild-tests/main_test.go

```diff
@@ -0,0 +1,343 @@
+	runComposerCLI(t, false, "compose", "types")
+	runComposerCLI(t, false, "compose", "status")
+	runComposerCLI(t, false, "compose", "list")
+	runComposerCLI(t, false, "compose", "list", "waiting")
+	runComposerCLI(t, false, "compose", "list", "running")
+	runComposerCLI(t, false, "compose", "list", "finished")
+	runComposerCLI(t, false, "compose", "list", "failed")
+	// runCommand(false, "compose", "log", UUID, [<SIZE>])
+	// runCommand(false, "compose", "cancel", UUID)
+	// runCommand(false, "compose", "delete", UUID)
+	// runCommand(false, "compose", "info", UUID)
+	// runCommand(false, "compose", "metadata", UUID)
+	// runCommand(false, "compose", "logs", UUID)
+	// runCommand(false, "compose", "results", UUID)
+	// runCommand(false, "compose", "image", UUID)
+	runComposerCLI(t, false, "blueprints", "list")
+	// runCommand(false, "blueprints", "show", BLUEPRINT,....)
+	// runCommand(false, "blueprints", "changes", BLUEPRINT,....)
+	// runCommand(false, "blueprints", "diff", BLUEPRINT, FROM/NEWEST, TO/NEWEST/WORKSPACE)
+	// runCommand(false, "blueprints", "save", BLUEPRINT,...)
+	// runCommand(false, "blueprints", "delete", BLUEPRINT)
+	// runCommand(false, "blueprints", "depsolve", BLUEPRINT,...)
+	// runCommand(false, "blueprints", "push", BLUEPRINT.TOML)
+	// runCommand(false, "blueprints", "freeze", BLUEPRINT,...)
+	// runCommand(false, "blueprints", "freeze", "show", BLUEPRINT,...)
+	// runCommand(false, "blueprints", "freeze", "save", BLUEPRINT,...)
+	// runCommand(false, "blueprints", "tag", BLUEPRINT)
+	// runCommand(false, "blueprints", "undo", BLUEPRINT, COMMIT)
+	// runCommand(false, "blueprints", "workspace", BLUEPRINT)
+	runComposerCLI(t, false, "modules", "list")
+	runComposerCLI(t, false, "projects", "list")
+	runComposerCLI(t, false, "projects", "info", "filesystem")
+	runComposerCLI(t, false, "projects", "info", "filesystem", "kernel")
+	runComposerCLI(t, false, "sources", "list")
+	// runCommand(false, "sources", "info", "fedora")
+	// runCommand(false, "sources", "info", "fedora", "fedora-updates")
+	// runCommand(false, "sources", "add" SOURCES.TOML)
+	// runCommand(false, "sources", "change" SOURCES.TOML)
+	// runCommand(false, "sources", "delete" SOURCES.TOML)
+	runComposerCLI(t, false, "status", "show")
+
+	// Full integration tests
+	testCompose(t, "ami")
+}
+
+func testCompose(t *testing.T, outputType string) {
+	tmpdir := NewTemporaryWorkDir(t, "osbuild-tests-")
+	log.Println(tmpdir.Path)
+	defer tmpdir.Close(t)
+
+	bp := blueprint.Blueprint{
+		Name:        "empty",
+		Description: "Test empty blueprint in toml format",
+	}
+	pushBlueprint(t, &bp)
+	defer deleteBlueprint(t, &bp)
+
+	runComposerCLI(t, false, "blueprints", "save", "empty")
+	//if _, err := os.Stat("empty.toml"); os.IsNotExist(err) {
+	//log.Fatalf("empty.toml doesn not exist: %v", err)
+	//}
+	_, err := os.Stat("empty.toml")
+	require.Falsef(t, os.IsNotExist(err), "empty.toml doesn not exist: %v", err)
```

Comment:
I think this should abort on any kind of error, can be permissions not only file doesn't exist. If I'm not mistaken we should require.Nil instead, no ?

Reply:
Good point, I agree. I changed that to `require.Nilf` and modified the error message to be more generic.